src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -81,20 +83,27 @@ private void createDirectories(Path targetDir) {
     @Override
     public void put(Collection<SinkRecord> records) {
         try {
-            for (SinkRecord sinkRecord : records) {
-                log.info("SinkRecord: {}", sinkRecord);
-                TopicPartition topicPartition = new TopicPartition(sinkRecord.topic(), sinkRecord.kafkaPartition());
-                PartitionWriter partition = partitionWriters.get(topicPartition);
-                Record record = Record.fromSinkRecord(sinkRecord);
-                log.info("Record: {}", record);
-                partition.append(record);
-                if(sinkRecord.kafkaOffset() % 100 == 0) {
-                    log.debug("Backed up Topic %s, Partition %d, up to offset %d", sinkRecord.topic(), sinkRecord.kafkaPartition(), sinkRecord.kafkaOffset());
-                }
-            }
-            // Todo: refactor to own worker. E.g. using the scheduler of MM2
+            List<TopicPartition> partitionsToSyncOffsets = records.stream()
```

Comment:
Hm... I think we need to actually retain this list between calls (using a set, only adding to it). Think about this scenario:

1. we write 10 messages to topic `foo`, but no-one consumes them, and then we stop writing.
2. they get backed up, but consumer group offsets are just empty `{}` (since we have no consumer)
3. *then* consumer group `cg` starts to consume them.
4. Since no new messages arrives at `foo`, we will never update the offsets for `cg`, since its no longer in the `partitionsToSyncOffsets`.

However, I'm not sure that will work either.
- If there is a rebalance between step 2 and 3 we can potentially loose the offsets.
- And if we just keep the set of `partitionsToSyncOffsets` forever, will we continue to sync offsets for topics we no longer own.
- Or does the above 2 effects counter eachother? 🤔

Reply:
Ah. Look at this: https://kafka.apache.org/0102/javadoc/org/apache/kafka/connect/sink/SinkTask.html#onPartitionsAssigned(java.util.Collection) Lets simply populate `partitionsToSyncOffsets` in that method!